crates/zizmor/src/audit/impostor_commit.rs

```diff
@@ -192,15 +300,147 @@ impl Audit for ImpostorCommit {
         };
 
         if self.impostor(uses)? {
-            findings.push(
-                Self::finding()
-                    .severity(Severity::High)
-                    .confidence(Confidence::High)
-                    .add_location(step.location().primary().annotated(IMPOSTOR_ANNOTATION))
-                    .build(step.action())?,
-            );
+            let mut finding_builder = Self::finding()
+                .severity(Severity::High)
+                .confidence(Confidence::High)
+                .add_location(step.location().primary().annotated(IMPOSTOR_ANNOTATION));
+
+            if let Some(fix) = self.create_impostor_fix(uses, step) {
+                finding_builder = finding_builder.fix(fix);
+            }
+
+            findings.push(finding_builder.build(step.action())?);
         }
 
         Ok(findings)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #[cfg(feature = "gh-token-tests")]
+    use insta::assert_snapshot;
+
+    #[cfg(feature = "gh-token-tests")]
+    use crate::models::AsDocument as _;
+
+    use super::*;
+    use crate::{models::workflow::Workflow, registry::input::InputKey};
+
+    #[cfg(feature = "gh-token-tests")]
+    #[test]
+    fn test_impostor_commit_fix_snapshot() {
+        // Test with a workflow that uses a commit hash that doesn't exist in the target repository
+        // We'll use actions/hello-world-javascript-action with a commit from actions/checkout
+        // This creates an impostor scenario: valid commit, wrong repository
+        let workflow_content = r#"
+name: Test Impostor Commit Fix
+on: push
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/hello-world-javascript-action@692973e3d937129bcbf40652eb9f2f61becf3332  # This is a commit from actions/checkout, not hello-world
+"#;
```

Comment:
Huh, I'm actually kind of surprised this works -- the audit wasn't really meant to detect "transposed" commits like this, but commits that are valid in the fork network of the parent repo. I guess those two scenarios look pretty similar from our API queries, though.

(No changes required, I'm just surprised this works!)